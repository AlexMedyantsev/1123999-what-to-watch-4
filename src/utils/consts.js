export const VideoPreview = {
  width: 280,
  height: 175,
  IS_MUTED: true,
  INTERVAL_IN_SEC: 1000,
};

export const PREVIEW_PLAY_DELAY = 2000;

export const MovieDetailsTabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const GENRES = {
  ALL: `All genres`,
  ADVENTURE: `Adventure`,
  COMEDIES: `Comedy`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Drama`,
  FANTASY: `Fantasy`,
  HORROR: `Horror`,
  KIDS_AND_FAMILY: `Kid's and Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci - Fi`,
  STORY: `Story`,
  THRILLERS: `Thriller`,
};

export const MAX_NUMBER_GENRES = 9;

export const FILTER_ALL_GENRES = `All Genres`;

export const SHOWING_MOVIES_COUNT_ON_START = 8;

export const SIMILAR_MOVIES_COUNT = 4;

export const MOVIES_ADDED_ON_BUTTON_CLICK = 8;

export const ESCAPE_KEY = 27;
