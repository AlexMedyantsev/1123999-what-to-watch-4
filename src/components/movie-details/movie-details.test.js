import React from "react";
import renderer from "react-test-renderer";
import MovieDetails from "./movie-details.jsx";

import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {GENRES} from "../../utils/consts.js";

const mockStore = configureStore([]);

const movie = [
  {
    image: `a`,
    posterSrc: `a`,
    bgSrc: `a`,
    title: `Name`,
    genre: `Genre`,
    year: 2014,
    score: 8.9,
    level: `level`,
    movieLink: `link`,
    previewLink: `preview-link`,
    scoresCount: 220,
    description: `description`,
    director: `director`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 22,
    link: `movie-page.html`,
    key: `1`,
    backgroundColor: `background_color`,
    isFavorite: `is_favorite`
  },
];

const movieAsObject = {
  image: `a`,
  posterSrc: `a`,
  bgSrc: `a`,
  title: `Name`,
  genre: `Genre`,
  year: 2014,
  score: 8.9,
  level: `level`,
  movieLink: `link`,
  previewLink: `preview-link`,
  scoresCount: 220,
  description: `description`,
  director: `director`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  runTime: 22,
  link: `movie-page.html`,
  key: `1`,
  backgroundColor: `background_color`,
  isFavorite: `is_favorite`
};

describe(`Movie Details Snapshot`, () => {
  it(`Should MovieDetails render correctly`, () => {
    const store = mockStore({
      genre: GENRES.ALL,
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <MovieDetails
              movie={movieAsObject}
              onMovieCardClick={() => {}}
              similarMovies={movie}
            />
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});


