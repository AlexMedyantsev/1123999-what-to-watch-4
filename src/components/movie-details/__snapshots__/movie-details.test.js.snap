// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage test Should MoviePage component render correctly 1`] = `
<div>
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Name"
          src="a"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign In
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Name
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Genre
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/2"
              onClick={[Function]}
              type="button"
            >
              <svg
                fill="none"
                height="19"
                viewBox="0 0 19 19"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clipRule="evenodd"
                  d="M0 0L19 9.5L0 19V0Z"
                  fill="#EEE5B5"
                  fillRule="evenodd"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="/login"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="a"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <div
            className="movie-card__desc"
          >
            <nav
              className="movie-nav movie-card__nav"
            >
              <ul
                className="movie-nav__list"
              >
                <li
                  className="movie-nav__item movie-nav__item--active"
                  onClick={[Function]}
                >
                  <a
                    className="movie-nav__link"
                    href="#"
                  >
                    Overview
                  </a>
                </li>
                <li
                  className="movie-nav__item false"
                  onClick={[Function]}
                >
                  <a
                    className="movie-nav__link"
                    href="#"
                  >
                    Details
                  </a>
                </li>
                <li
                  className="movie-nav__item false"
                  onClick={[Function]}
                >
                  <a
                    className="movie-nav__link"
                    href="#"
                  >
                    Reviews
                  </a>
                </li>
              </ul>
            </nav>
            <div
              className="movie-rating"
            >
              <div
                className="movie-rating__score"
              >
                8.9
              </div>
              <p
                className="movie-rating__meta"
              >
                <span
                  className="movie-rating__level"
                >
                  level
                </span>
                <span
                  className="movie-rating__count"
                >
                  220
                   ratings
                </span>
              </p>
            </div>
            <div
              className="movie-card__text"
            >
              description
              <p
                className="movie-card__director"
              >
                <strong>
                  Director: 
                  director
                </strong>
              </p>
              <p
                className="movie-card__starring"
              >
                <strong>
                  Starring: 
                  Bill Murray, Edward Norton, Jude Law, Willem Dafoe, Saoirse Ronan
                   and other
                </strong>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseOut={[Function]}
          >
            <video>
              Sorry, this preview is not available in your browser
            </video>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Name
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseOut={[Function]}
          >
            <video>
              Sorry, this preview is not available in your browser
            </video>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Name
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <div
          className="logo__link logo__link--light"
        >
          <div
            className="logo"
          >
            <a
              className="logo__link"
              href="/"
              onClick={[Function]}
            >
              <span
                className="logo__letter logo__letter--1"
              >
                W
              </span>
              <span
                className="logo__letter logo__letter--2"
              >
                T
              </span>
              <span
                className="logo__letter logo__letter--3"
              >
                W
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
